optimization/+DoseConstraints/matRad_ObjectiveConstraint.m

```diff
@@ -0,0 +1,111 @@
+classdef matRad_ObjectiveConstraint < DoseConstraints.matRad_DoseConstraint
+    % matRad_ObjectiveConstraint implements a wrapper function that turns 
+    % dose objectives into constraints
+    %
+    % use log sum exp approximation, see appendix A in
+    % http://scitation.aip.org/content/aapm/journal/medphys/41/8/10.1118/1.4883837
+    %
+    % %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+    %
+    % Copyright 2020 the matRad development team. 
+    % 
+    % This file is part of the matRad project. It is subject to the license 
+    % terms in the LICENSE file found in the top-level directory of this 
+    % distribution and at https://github.com/e0404/matRad/LICENSES.txt. No part 
+    % of the matRad project, including this file, may be copied, modified, 
+    % propagated, or distributed except according to the terms contained in the 
+    % LICENSE file.
+    %
+    % %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+    properties (Constant)
+        name = 'Objective Constraint';
+        parameterTypes = {'objFunc'};
+        parameterNames = {'f^{max}'};    
+    end
+
+    properties
+        objective;
+        parameters = {1e-5};
+        epsilon = 1e-3; %
+    end
+    
+    methods (Access = public)
+        function obj = matRad_ObjectiveConstraint(objective,maxObj,epsilon)
+            
+            %check if objective is a struct and a DoseObjective or Constraint (for init from constraint)
+            if isstruct(objective) && ~isempty(strfind(objective.className,'DoseObjectives'))
+                objective =  matRad_DoseOptimizationFunction.createInstanceFromStruct(objective);
+            end
+
+            if nargin == 1 && isstruct(objective)
+                initFromStruct = true;
+                inputStruct = objective;
+            else
+                initFromStruct = false;
+                inputStruct = [];
+            end
+
+            obj@DoseConstraints.matRad_DoseConstraint(inputStruct);
```

Comment:
for readability i would change this 'obj' named instance to something else, 'doseConstraint'